src/target/ext/host_io.rs

```diff
@@ -140,31 +200,38 @@ define_ext!(HostIoCloseOps, HostIoClose);
 
 /// Nested Target Extension - Host I/O pread operation.
 pub trait HostIoPread: HostIo {
-    /// Read data from the open file corresponding to fd. Up to count bytes will
-    /// be read from the file, starting at offset relative to the start of the
-    /// file. The target may read fewer bytes; common reasons include packet
-    /// size limits and an end-of-file condition. The number of bytes read is
-    /// returned. Zero should only be returned for a successful read at the end
-    /// of the file, or if count was zero.
-    fn pread(
+    /// Read data from the open file corresponding to fd.
+    ///
+    /// Up to count bytes will be read from the file, starting at offset
+    /// relative to the start of the file.
+    ///
+    /// The target may read fewer bytes; common reasons include packet size
+    /// limits and an end-of-file condition.
+    ///
+    /// The number of bytes read is returned. Zero should only be returned for a
+    /// successful read at the end of the file, or if count was zero.
+    fn pread<'a>(
         &mut self,
         fd: i32,
         count: <Self::Arch as Arch>::Usize,
         offset: <Self::Arch as Arch>::Usize,
-        output: HostIoOutput<'_>,
-    ) -> TargetResult<(), Self>;
+        output: PreadOutput<'a>,
+    ) -> TargetResult<PreadToken<'a>, Self>;
 }
 
 define_ext!(HostIoPreadOps, HostIoPread);
 
 /// Nested Target Extension - Host I/O pwrite operation.
 pub trait HostIoPwrite: HostIo {
-    /// Write data (a binary buffer) to the open file corresponding to fd. Start
-    /// the write at offset from the start of the file. Unlike many write system
-    /// calls, there is no separate count argument; the length of data in the
-    /// packet is used. ‘vFile:pwrite’ returns the number of bytes written,
-    /// which may be shorter than the length of data, or -1 if an error
-    /// occurred.
+    /// Write data (a binary buffer) to the open file corresponding to fd.
+    ///
+    /// Start the write at offset from the start of the file.
+    ///
+    /// Unlike many write system calls, there is no separate count argument; the
+    /// length of data in the packet is used.
+    ///
+    /// ‘vFile:pwrite’ returns the number of bytes written, which may be shorter
+    /// than the length of data, or -1 if an error occurred.
```

Comment:
(reminder to update docs when `HostIoResult` error handling is implemented)

Reply:
Example?

Reply:
This is a great example. The docs currently talk about "returning -1 if an error occured", when that isn't applicable anymore.